galileo/legged-model/src/LeggedInterface.cpp

```diff
@@ -0,0 +1,203 @@
+
+namespace galileo
+{
+    namespace legged
+    {
+        LeggedInterface::LeggedInterface()
+        {
+            surfaces_ = std::make_shared<galileo::legged::environment::EnvironmentSurfaces>();
+        }
+
+        void LeggedInterface::LoadModel(std::string model_file_location, std::vector<std::string> end_effector_names)
+        {
+            std::string *end_effector_names_array = new std::string[end_effector_names.size()];
+
+            for (size_t i = 0; i < end_effector_names.size(); ++i)
+            {
+                end_effector_names_array[i] = end_effector_names[i];
+            }
+
+            robot_ = std::make_shared<LeggedBody>(model_file_location, end_effector_names.size(), end_effector_names_array);
+            states_ = robot_->si;
+        }
+
+        void LeggedInterface::LoadParameters(std::string parameter_file_location)
+        {
+            // Hardcoded parameters for now
+            // Load the parameters from the given parameter file.
+            opts_["ipopt.fixed_variable_treatment"] = "make_constraint";
+            opts_["ipopt.max_iter"] = 10;
+
+            cost_params_.R_diag = Eigen::VectorXd(states_->nu);
+            cost_params_.R_diag << 1e-3, 1e-3, 1e-3,                        /*First contact wrench error weights*/
+                1e-3, 1e-3, 1e-3,                                           /*Second contact wrench error weights*/
+                1e-3, 1e-3, 1e-3,                                           /*Third contact wrench error weights*/
+                1e-3, 1e-3, 1e-3,                                           /*Fourth contact wrench error weights*/
+                10., 10., 10., 10., 10., 10., 10., 10., 10., 10., 10., 10.; /*Joint acceleration error weights*/
+
+            cost_params_.Q_diag = Eigen::VectorXd(states_->ndx);
+            cost_params_.Q_diag << 15., 15., 30., 5., 10., 10.,             /*Centroidal momentum error weights*/
+                0., 0., 0., 0., 0., 0.,                                     /*Rate of Centroidal momentum error weights*/
+                500., 500., 500., 0.1, 0.1, 0.1,                            /*Floating base position and orientation (exponential coordinates) error weights*/
+                20., 20., 20., 20., 20., 20., 20., 20., 20., 20., 20., 20., /*Joint position error weights*/
+                0., 0., 0., 0., 0., 0.,                                     /*Floating base velocity error weights*/
+                0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.;             /*Joint velocity error weights*/
+        }
+
+        void LeggedInterface::CreateProblemData(const T_ROBOT_STATE &initial_state, const T_ROBOT_STATE &target_state)
+        {
+            casadi::Function L, Phi;
+
+            CreateCost(initial_state, target_state, L, Phi);
+
+            std::shared_ptr<opt::GeneralProblemData> gp_data = std::make_shared<opt::GeneralProblemData>(robot_->fint, robot_->fdif, L, Phi);
```

Comment:
This should be fdiff

Reply:
has that change been merged yet? I was under the impression it was on another branch